src/humancellatlas/data/metadata/api.py

```diff
@@ -107,20 +107,91 @@ def __init__(self, entity: LinkedEntity, other_entity: Entity, forward: bool) ->
                          other_entity.address)
 
 
+@dataclass(init=False)
+class ProjectPublication(object):
+    """
+    Project Publication
+
+    Schema: http://schema.staging.data.humancellatlas.org/module/project/5.2.2/publication
+    """
+    authors: List[str]
+    publication_title: str
+    doi: str
+    pmid: int
+    publication_url: str
+
+    def __init__(self, json: JSON) -> None:
+        self.authors = json.get('authors')
+        self.publication_title = json.get('publication_title')
+        self.doi = json.get('doi')
+        self.pmid = json.get('pmid')
+        self.publication_url = json.get('publication_url')
+
+    def __hash__(self):
+        return hash('::'.join(['/'.join(self.authors), self.publication_url,
+                               self.doi or 'undefined_doi', str(self.pmid or 'undefined_pmid'),
```

Comment:
I just tried `str('undefined_pmid')` on Jupyter and it seems to be fine.

<img width="648" alt="screen shot 2018-09-26 at 11 42 38 am" src="https://user-images.githubusercontent.com/226612/46091727-61c45980-c181-11e8-9e5f-1a926936d6e6.png">

Could you describe how it would crash?